stl/inc/memory

```diff
@@ -405,6 +871,48 @@ _NoThrowFwdIt uninitialized_value_construct_n(_NoThrowFwdIt _First, const _Diff
     return _First;
 }
 
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // VARIABLE ranges::uninitialized_value_construct_n
+    class _Uninitialized_value_construct_n_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <_No_throw_forward_iterator _It>
+            requires default_initializable<iter_value_t<_It>>
+        _It operator()(_It _First, const iter_difference_t<_It> _Count_raw) const {
+            // clang-format on
+            using _Ty                                       = iter_value_t<_It>;
+            _Algorithm_int_t<iter_difference_t<_It>> _Count = _Count_raw;
+            if (_Count > 0) {
+                auto _UFirst = _Get_unwrapped_n(_STD move(_First), _Count);
+                if constexpr (_Use_memset_value_construct_v<_It>) {
+                    _Seek_wrapped(_First, _RANGES _Zero_range(_UFirst, _UFirst + _Count));

```

Comment:
Should we just update `_UFirst` here, and move the `_Seek_wrapped` call out of the branching structure?

Reply:
That would require an additional move construction for the `_Backout` case